AK/HashTable.h

```diff
@@ -15,6 +15,7 @@
 namespace AK {
 
 enum class HashSetResult {
+    Failed = 0,
```

Comment:
`Failed` doesn't really say anything. Did we fail to allocate, or did something else fail? Right now, we don't return a result for any other failure, but I feel like this could be more descriptive nevertheless.

Reply:
Setting a HashTable can only fail one way as far as I know

The only other thing I can think of are collisions and we just rehash in these cases